nncf/nncf_network.py

```diff
@@ -747,3 +749,18 @@ def get_MACs_in_model(self):
 
     def get_input_infos(self) -> List[ModelInputInfo]:
         return deepcopy(self.input_infos)
+
+    @staticmethod
+    def collect_eval_only_ops_exec_context(model: nn.Module, graph_builder) -> List[str]:
+        """
+        Returns scopes of the modules which are executed in evaluation mode only.
+        Model is set to eval mode.
```

Comment:
Should avoid such side effects in the future and keep the model in the state it arrived in.